batch/batch/front_end/front_end.py

```diff
@@ -893,6 +894,18 @@ async def _create_jobs(userdata: dict, job_specs: dict, batch_id: int, update_id
         resources['storage_gib'] = storage_gib
         resources['preemptible'] = preemptible
 
+        regions = spec.get('regions')
+        if regions is not None:
+            valid_regions = set(app['regions'].keys())
+            invalid_user_regions = set(regions).difference(valid_regions)
+            if invalid_user_regions:
+                raise web.HTTPBadRequest(
+                    reason=f'invalid regions specified: {invalid_user_regions}. Choose from {valid_regions}'
+                )
+            if len(regions) == 0:
+                raise web.HTTPBadRequest(reason='must choose at least one region or use "None" to specify all regions')
```

Comment:
We should avoid messages that specifically reference Python in HTTP responses. In general our users (e.g. Jonah) might be calling us from some other language or cooking up JSON by hand. I'd just go with something like "regions must not be the empty array".